pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -85,14 +101,31 @@ func (a *GCPActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *hi
 		return nil, false, errors.New("MachinePool is not for GCP")
 	}
 
-	leaseChar, proceed, err := a.obtainLease(pool, cd, logger)
+	leases := &hivev1.MachinePoolNameLeaseList{}
+	err := a.client.List(context.TODO(), leases, client.InNamespace(pool.Namespace),
+		client.MatchingLabels(map[string]string{
+			constants.ClusterDeploymentNameLabel: cd.Name,
+		}))
 	if err != nil {
-		logger.WithError(err).Log(controllerutils.LogLevel(err), "error obtaining pool name lease")
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "error fetching machinepoolleases")
 		return nil, false, err
 	}
 
-	if !proceed {
-		return nil, false, nil
+	poolName := pool.Spec.Name
+
+	// If leases are required by the cluster version or existing "w" worker machinesets in the cluster or are already
+	// being used as indicated by the existence of MachinePoolLeases, then use leases for determining the machine pool
+	// name.
+	if a.leasesRequired || len(leases.Items) > 0 {
```

Comment:
Could you add some logging here so we can see what the controller decided to do for this cluster and why. I'd kind of like to know if it was because the version required leases or if it didn't but we found them already in use.